### ISSUE TITLE
Feature request -  hardlinks count or add a pre/post script option

### ISSUE THREAD

**Issue by Jorman** [jerrymakesjelly/autoremove-torrents#146]
Hi, I'm making a bash script to better handle my torrents, the idea is pretty simple

The idea is:
I delete only torrents (in certain categories) but only if seed time is enough and if hardlink count is more than 1.
Usually I delete torrent only after x time and/or seeding time, but just came into my mind that I make hardlinks and maybe this "copy" stay in place for months, so, because is one hardlink don't take extra space and so, why delete it? is only one torrent that can help in seed ratio, I can delete it when the hardlinks count is 1 and all remaining conditions are good. I hope I explained my idea.

So for this reason I created a bash script that do all this pre-check, (only for selected categories) and after that call the autoremove-torrents to clean all remaining categories.

Or do you think that is possible to check hardlinks count inside autoremove-torrents? Make sense for you?

J

**Comment by jerrymakesjelly** [comment]
I think I get your point. Do you mean that you need a mechanism, which checks the hardlink count before deleting torrents, and only deletes torrent when the hardlink count of the torrent equals to 1?

If so, I prefer to provide a mechanism to check the hardlink count in autoremove-torrents, instead of pre/post script option.

**Comment by Jorman** [comment]
Yes, you got the point, equal to 1 and with the same logic, like seed time > X or all others function, like now.
On qBittorren you already have the file, I made this for qBittorrent, but is a script! I use systemd timers so I can start it before your autoremove-torrents easily.
https://github.com/Jorman/Scripts/blob/master/qBittorrentHardlinksChecker.sh
qBittorrent save the path so I can easily check if the file have one or more hardlinks, maybe all others client have the saving path information from the API?

**Comment by GlassedSilver** [comment]
Any update on this? I find this EXTREMELY interesting, because it can help keep torrents healthy without having to manually figure out which ones are only stored anymore for seeding when you possibly already switched them out for something better for yourself, now wasting space.

**Comment by jz1** [comment]
> Any update on this? I find this EXTREMELY interesting, because it can help keep torrents healthy without having to manually figure out which ones are only stored anymore for seeding when you possibly already switched them out for something better for yourself, now wasting space.

Just made a draft pull, you can test if you are interested for it https://github.com/jerrymakesjelly/autoremove-torrents/pull/198

**Comment by Jorman** [comment]
@GlassedSilver I made a small script, in bash and python, to help with this, then an external script with some other functionality
https://github.com/Jorman/Scripts
Just in case the pull will not be enclosed on this

**Comment by jz1** [comment]
> @GlassedSilver I made a small script, in bash and python, to help with this, then an external script with some other functionality https://github.com/Jorman/Scripts Just in case the pull will not be enclosed on this

i closed the pull again, figured that i only added it for qbittorent, but that it should actually also support Transmission/μTorrent/Deluge. Proberly easy to add since the client support only needs to be added to grab the path link but otherways someone can just edit and reupload it and create a new pull with it.

**Comment by Jorman** [comment]
@jz1 Yep, I think this's the tricky point for this idea, maybe is not so easy for others client? I don't know, I only use qBittorrent right now, so for this I created the script, for my purpose and then shared, maybe can be useful for someone
